gabriel-portfolio/portfolio/src/main/webapp/script.js

```diff
@@ -13,33 +13,73 @@
 // limitations under the License.
 
 /**
- * Adds a random greeting to the page.
+ * Adds a random fact to the page.
  */
-function addRandomGreeting() {
-  const greetings =
+function addRandomFact() {
+  const randomFacts =
       ['I can solve a rubiks cube in less than a minute',
        'I speak english and spanish fluently',
        'Both of my parents are from Cuba',
        'My favorite food is fried fish'];
 
   // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+  const randomFact = randomFacts[Math.floor(Math.random() * randomFacts.length)];
 
   // Show the hide-message button
-  const hide = document.getElementById("hide-message");
+  const hide = document.getElementById("hide-random-fact");
   hide.style.visibility = "visible";
 
   // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+  const randomFactContainer = document.getElementById('random-fact-container');
+  randomFactContainer.innerText = randomFact;
 }
 
-function hideMessage() {
+/*
+  hides the random fact
+*/
+function hideRandomFact() {
   // Set the message to an empty string
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = "";
+  const randomFactContainer = document.getElementById('random-fact-container');
+  randomFactContainer.innerText = "";
 
   // Hide the hide-message button
-  const button = document.getElementById("hide-message");
+  const button = document.getElementById("hide-random-fact");
   button.style.visibility = "hidden";
 }
+
+/*
+  uses the fetch() function to get the greeting text from /get-greeting
+*/
+async function getGreeting() {
+  const response = await fetch('/get-greeting');
+  const quote = await response.text();
+  document.getElementById('greeting-container').innerText = quote;
+}
+
+/* 
+  Creates an <li> element containing text 
+*/
+function createListElement(text) {
+  const liElement = document.createElement('li');
+  liElement.innerText = text;
+  return liElement;
+}
+
+/*
+  uses fetch() to get comments from /data
+*/
+async function getComments() {
+  const response = await fetch('/data');
+  const comments = await response.json();
+  console.log(comments);
+  const commentList = document.getElementById('comments-container');
+  commentList.innerHTML = '';
+  // loop through the strings in the json object
+  const maxNumberOfComments = document.getElementById("maxNumberOfComments").value;
+  console.log(maxNumberOfComments);
+  var i=0;
+  while(comments[i] && i < maxNumberOfComments) {
```

Comment:
Not blocking for this review, but instead of enforcing the limit here, you should consider passing the limit with the fetch request.

Reply:
I wanted to keep the the number of comments feature to be separate from submitting a comment. I thought sending the limit to the javascript function that prints the comments would be best. How exactly could I send the limit with the fetch request?

Reply:
I was wondering if it would be possible to retrieve the comments from the DataServlet and then directly limit the number of comments by passing a count parameter (I think similar to what Lu mentioned in the comment below), but this definitely works for now.